src/platform/backends/shared/base_snapshot.h

```diff
@@ -37,6 +37,7 @@ class BaseSnapshot : public Snapshot
 public:
     BaseSnapshot(const std::string& name,
                  const std::string& comment,
+                 const std::string& instance_id,
```

Comment:
Can we call this `cloud_init_id` instead? `instance_id` is fine in the context of cloud init code, but here it is a little enigmatic. Readers would wonder "how does this differ from the name?" and they would have to read code to understand. Better make the name immediately obvious (here and in accessors).

Reply:
Yep, sounds good to me.